sys/signal.c

```diff
@@ -0,0 +1,221 @@
+#include <signal.h>
+#include <thread.h>
+#include <malloc.h>
+#include <stdc.h>
+#include <errno.h>
+#include <sysent.h>
+#include <systm.h>
+
+static MALLOC_DEFINE(sig_pool, "signal handlers pool");
+
+static sighandler_t *signal_default_actions[SIG_LAST] = {
+    [SIGINT] = SIG_TERM,  [SIGILL] = SIG_TERM,  [SIGFPE] = SIG_TERM,
+    [SIGABRT] = SIG_TERM, [SIGSEGV] = SIG_TERM, [SIGKILL] = SIG_TERM,
+    [SIGCHLD] = SIG_IGN,  [SIGUSR1] = SIG_TERM, [SIGUSR2] = SIG_TERM,
+};
+
+static const char *signal_names[SIG_LAST] = {
+    [SIGINT] = "SIGINT",   [SIGILL] = "SIGILL",   [SIGABRT] = "SIGABRT",
+    [SIGFPE] = "SIGFPE",   [SIGSEGV] = "SIGSEGV", [SIGKILL] = "SIGKILL",
+    [SIGTERM] = "SIGTERM", [SIGCHLD] = "SIGCHLD", [SIGUSR1] = "SIGUSR1",
+    [SIGUSR2] = "SIGUSR2",
+};
+
+void sighand_init() {
+  /* TODO: A lot of kernel components have a init function whose only purpose is
+     to perform a kmalloc_init. Maybe we could collect them with a linker set,
+     and initialize all pools together? */
+  kmalloc_init(sig_pool, 2, 2);
```

Comment:
I'm growing convinced that after pool allocator is merged we'll switch to single memory pool for `kmalloc`. I find passing memory pool to `kmalloc` functions cumbersome. NetBSD counterpart [malloc(9)](http://netbsd.gw.com/cgi-bin/man-cgi?malloc+9+NetBSD-current) is deprecated in favour of system-wide memory allocator: 

> These interfaces are being obsoleted and their new use is discouraged. For new code, use kmem(9) for variable-sized or one-time allocations and pool_cache(9) for frequent fixed-size allocations instead.